src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -50,28 +60,177 @@
+      List<IexHistoricalPrices> outList = new ArrayList<>();
+      for (String dateVal : datesToLookThrough) {
+        List<IexHistoricalPrices> out = getHistoricalPriceBySymbolAndDate(symbol, dateVal);
+        if (!out.isEmpty()) {
+          outList.add(out.get(0));
+        }
+      }
+      return outList;
     }
   }
 
+  /**
+   * Converts a range string into a list of dates (string) in YYYYMMDD format.
+   * Will ignore weekend dates. Later, will need to also remove bank holidays.
+   *
+   * @param range The range of time we are interested. Input options are:
+   *      - "5y", "2m", "7d": x number of years(y)/months(m)/days(d)
+   *      - "ytd": year to date
+   *
+   * @return a list of valid date strings (YYYYMMDD) in the range
+   */
+  public List<String> rangeToDateList(final String range) {
+    LocalDate endDate = LocalDate.now();
+    LocalDate startDate;
+
+    if (range.equalsIgnoreCase("ytd")) {
+      startDate = LocalDate.ofYearDay(endDate.getYear(), 1);
+    } else {
+      // Confirms range is valid entry
+      String pattern = "\\d+[dmyDMY]";
+      Pattern p = Pattern.compile(pattern);
+      Matcher m = p.matcher(range);
+      if (!m.matches()) {
+        throw new IllegalArgumentException("Improper Range Provided.");
+      }
+      // Parses string to correct start date
+      String tp = range.substring(range.length() - 1);
+      int numOfTime = Integer.parseInt(range.substring(0,range.length()-1));
+      if (tp.equalsIgnoreCase("d")) {
+        startDate = endDate.minusDays(numOfTime);
+      } else if (tp.equalsIgnoreCase("m")) {
+        startDate = endDate.minusMonths(numOfTime);
+      } else {
+        startDate = endDate.minusYears(numOfTime);
+      }
+    }
+
+    // Predicate for checking on weekends
+    Predicate<LocalDate> isWeekend = date -> date.getDayOfWeek() == DayOfWeek.SATURDAY
+        || date.getDayOfWeek() == DayOfWeek.SUNDAY;
+
+    // List of Business Days
+    List<LocalDate> businessDays = startDate.datesUntil(endDate)
+        .filter(isWeekend.negate())
+        .collect(Collectors.toList());
+
+    // Convert to list of strings in proper format
+    List<String> dateList = new ArrayList<>();
+    for (LocalDate lD : businessDays) {
+      dateList.add(lD.format(DateTimeFormatter.ofPattern("yyyyMMdd")));
+    }
+    return dateList;
+  }
+
+  /**
+   * Chooses whether to get the historical price for the symbol from the repository or Iex.
+   * Precedence is always given to the repository, unless is not found within it.
+   *
+   * @param symbol Stock symbol to get the historical price for.
+   * @param date Input date (formatted YYYYMMDD) to get price data for.
+   *      Note: We will convert this into YYYY-MM-DD format for repository search.
+   *      - Format of Iex API output
+   * @return a list (length = 1) of IexHistoricalPrices objects.
+   */
+  public List<IexHistoricalPrices> getHistoricalPriceBySymbolAndDate(
```

Comment:
The way this is coded, won't we call to IEX and to the DB even on weekends when we know there wouldn't be any prices in either?

Reply:
As it's coded now, this would only occur when we specifically pass in a specific date that is on the weekend (the user calls for a specific weekend date). The rangeToDateList method automatically removes the weekends before passing in the list of dates to this function.

Reply:
I corrected this behavior as well, so now the system will check if a specific date is a weekend before passing it forward into getHistoricalPriceBySymbolAndDate as well as if we have a range.